core/ledger/kvledger/txmgmt/privacyenabledstate/namespace_helper.go

```diff
@@ -0,0 +1,121 @@
+
+package privacyenabledstate
+
+import (
+	"github.com/hyperledger/fabric-protos-go/ledger/queryresult"
+	commonledger "github.com/hyperledger/fabric/common/ledger"
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/statedb"
+	"github.com/pkg/errors"
+)
+
+// constructStatedbNamespaces constructs all the namespaces for a channel, including the following:
+// - special namespaces ("", _lifecycle, lscc)
+// - namespaces for implicit collections,
+// - namespaces for deployment chaincodes and explicit collections.
+func constructStatedbNamespaces(
+	channelName string,
+	vdb statedb.VersionedDB,
+	channelInfoProvider channelInfoProvider,
+	deployedccInfoProvider ledger.DeployedChaincodeInfoProvider,
+) ([]string, error) {
+	retNamespaces := []string{}
+
+	// add special namespaces: "" and deployedccInfoProvider.Namespaces()
+	retNamespaces = append(retNamespaces, "")
+	for _, ns := range deployedccInfoProvider.Namespaces() {
+		retNamespaces = append(retNamespaces, ns)
+	}
+
+	// add namespaces for implicit collections
+	mspids, err := channelInfoProvider.GetAllMSPIDs()
+	if err != nil {
+		return nil, err
+	}
+	implicitCollNames := deployedccInfoProvider.ImplicitCollectionNames(mspids)
+	for _, implicitCollName := range implicitCollNames {
+		retNamespaces = append(retNamespaces, derivePvtDataNs("_lifecycle", implicitCollName))
+		retNamespaces = append(retNamespaces, deriveHashedDataNs("_lifecycle", implicitCollName))
+	}
+
+	// add namespaces for chaincodes and explicit collections
+	chaincodesInfo, err := deployedccInfoProvider.AllChaincodesInfo(channelName, &simpleQueryExecutor{vdb})
+	if err != nil {
+		return nil, err
+	}
+	for _, ccInfo := range chaincodesInfo {
+		retNamespaces = append(retNamespaces, ccInfo.Name)
+		if ccInfo.ExplicitCollectionConfigPkg == nil {
+			continue
+		}
+		for _, config := range ccInfo.ExplicitCollectionConfigPkg.Config {
+			collConfig := config.GetStaticCollectionConfig()
+			if collConfig != nil {
+				retNamespaces = append(retNamespaces, derivePvtDataNs(ccInfo.Name, collConfig.Name))
+				retNamespaces = append(retNamespaces, deriveHashedDataNs(ccInfo.Name, collConfig.Name))
+			}
+		}
+	}
+
+	return retNamespaces, nil
+}
+
+// simpleQueryExecutor implements ledger.SimpleQueryExecutor interface
+type simpleQueryExecutor struct {
+	vdb statedb.VersionedDB
+}
+
+func (sqe *simpleQueryExecutor) GetState(ns string, key string) ([]byte, error) {
+	versionedValue, err := sqe.vdb.GetState(ns, key)
+	if err != nil {
+		return nil, err
+	}
+	var value []byte
+	if versionedValue != nil {
+		value = versionedValue.Value
+	}
+	return value, nil
+}
+
+func (sqe *simpleQueryExecutor) GetStateRangeScanIterator(ns string, startKey string, endKey string) (commonledger.ResultsIterator, error) {
+	dbItr, err := sqe.vdb.GetStateRangeScanIterator(ns, startKey, endKey)
+	if err != nil {
+		return nil, err
+	}
+	itr := &resultsItr{ns: ns, dbItr: dbItr}
+	return itr, nil
+}
```

Comment:
```suggestion
func (sqe *simpleQueryExecutor) GetStateRangeScanIterator(ns string, startKey string, endKey string) (commonledger.ResultsIterator, error) {
	dbItr, err := sqe.vdb.GetStateRangeScanIterator(ns, startKey, endKey)
	return dbItr.(commonledger.ResultsIterator), err
}
```
Wouldn't this work?

At least for the range query, this redirection is bit forced due to duplication of the `ResultIterator` interface. In the `statedb.go`, we have a `ResultIterator` that is used by the each statedb implementation. Then, in the `common/ledger/ledger_interface.go`, we have the same interface. Due to the mixed usage, we couldn't simply reuse the iterator present at the `VersionedDB`. 

To address this correctly, we need to refactor the existing code and remove the duplicate interfaces used for range query iterators.